istresearch/traptor
# Pulse 3618: Add limit messages to redis

This PR adds the ability for Traptor to log limit message counts into Redis using an expiring key. The expiration time is the same as the rolling time window currently configured.

**Testing**

Get the code

- `git clone git@github.com:istresearch/traptor.git`
- `git checkout PULSE-3618-add-limit-messages-to-redis`

Set up your environment

- Create and activate a virtual environment (venv or Anaconda)
- Rename traptor.env.sample to traptor.env and fill in all fields. Some defaults are provided.
- Install the requirements: `pip install -r requirements.txt`

Run the tests

- In the root directory: `python -m pytest -vv --cache-clear --cov=traptor --cov-report html`

All tests should pass.
